RubberduckTests/CodeExplorer/CodeExplorerTests.cs

```diff
@@ -2872,48 +1257,440 @@ public void CompareByNodeType_FoldersAreSortedByName()
             {
-                EnableExperimentalFeatures = new List<ExperimentalFeatures>
+                _generalSettingsProvider.Setup(s => s.Create()).Returns(_generalSettings);
+                _windowSettingsProvider.Setup(s => s.Create()).Returns(WindowSettings);
+                _configLoader.Setup(c => c.LoadConfiguration()).Returns(GetDefaultUnitTestConfig());
+
+                SaveDialog = new Mock<ISaveFileDialog>();
+                SaveDialog.Setup(o => o.OverwritePrompt);
+
+                OpenDialog = new Mock<IOpenFileDialog>();
+                OpenDialog.Setup(o => o.AddExtension);
+                OpenDialog.Setup(o => o.AutoUpgradeEnabled);
+                OpenDialog.Setup(o => o.CheckFileExists);
+                OpenDialog.Setup(o => o.Multiselect);
+                OpenDialog.Setup(o => o.ShowHelp);
+                OpenDialog.Setup(o => o.Filter);
+                OpenDialog.Setup(o => o.CheckFileExists);
+
+                FolderBrowser = new Mock<IFolderBrowser>();
+                _browserFactory
+                    .Setup(m => m.CreateFolderBrowser(It.IsAny<string>(), true,
+                        @"C:\Users\Rubberduck\Documents\Subfolder")).Returns(FolderBrowser.Object);
+            }
+
+            public MockedCodeExplorer(string code) : this(ProjectType.HostProject, ComponentType.StandardModule, code) { }
+
+            //public MockedCodeExplorer(ProjectType projectType, ComponentType componentType = ComponentType.StandardModule, string code = "")
+            //    : this(projectType, new[] { componentType }, new[] { code }) { }
+
+            public MockedCodeExplorer(ProjectType projectType, ComponentType componentType = ComponentType.StandardModule, string code = "") : this()
+            {
+                var builder = new MockVbeBuilder();
+                var project = builder.ProjectBuilder("TestProject1", ProjectProtection.Unprotected, projectType)
+                    .AddComponent("TestModule", componentType, code);
+
+                VbComponents = project.MockVBComponents;
+                VbComponent = project.MockComponents.First();
+                VbProject = project.Build();
+                Vbe = builder.AddProject(VbProject).Build();
+
+                var projectRepository = new ProjectsRepository(Vbe.Object);
+
+                State = new RubberduckParserState(Vbe.Object, projectRepository, new DeclarationFinderFactory(), MockVbeEvents.CreateMockVbeEvents(Vbe).Object);
+
+                var removeCommand = new RemoveCommand(SaveDialog.Object, MessageBox.Object, State.ProjectsProvider);
+
+                ViewModel = new CodeExplorerViewModel(new FolderHelper(State, Vbe.Object), State, removeCommand,
+                    _generalSettingsProvider.Object, _windowSettingsProvider.Object, _uiDispatcher.Object, Vbe.Object, null);
+
+                var parser = MockParser.Create(Vbe.Object, State, projectRepository);
+                parser.Parse(new CancellationTokenSource());
```

Comment:
Do we really have to set this up here explicitly? Why not use `MockVBAParser.Create(Vbe)` (Might need a slight correction to base the `MockVbeEvents` on the VBE passed in. And then get the state from the parser?

Reply:
Not really. This entire file was acting "tweaky" enough that I stopped refactoring when I got all the tests passing.  Changing to `MockVBAParser.Create(Vbe)` is a good idea though.

Reply:
This is fixed in the CE PR that's forthcoming.